### PR TITLE
[Enhancement](regression-test) support exclude suite/group/directory

### PR DESCRIPTION
**Pull request by 924060929**
# Proposed changes

regression testing framework support skip some suite/group/directory

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No)
3. Has document been added or modified: (Yes)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (Yes)

### CODE REVIEW

File: run-regression-test.sh [hunk 1]
```diff
@@ -56,7 +36,11 @@ Usage: $0 <shell_options> <framework_options>
   Optional framework_options:
      -s                                run a specified suite
      -g                                run a specified group
+     -d                                run a specified directory
      -h                                **print all framework options usage**
+     -xs                               exclude the specified suite
+     -xg                               exclude the specified group
+     -xd                               exclude the specified directory
```
Comment (pingchunzhang) [inline run-regression-test.sh:43]: by the way, what does x mean, why did not choose -es -eg -ed to exclude the specific suite/group/directory

Comment (924060929) [reply run-regression-test.sh:43]: `es` maybe conflict with elasticsearch, so i change to `eXclude`

File: regression-test/framework/src/main/groovy/org/apache/doris/regression/ConfigOptions.groovy [hunk 2]
```diff
@@ -132,6 +144,15 @@ class ConfigOptions {
...
                 .desc("the suite group to be test")
                 .build()
+        excludeGroupsOpt = Option.builder("xg")
+                .argName("excludeGroups")
+                .required(false)
+                .hasArg(true)
+                .optionalArg(true)
+                .type(String.class)
+                .longOpt("groups")
+                .desc("the suite group will not be tested")
```
[the remark this answers is not in the document]
Comment (924060929) [reply regression-test/framework/src/main/groovy/org/apache/doris/regression/ConfigOptions.groovy:154]: done

File: regression-test/framework/src/main/groovy/org/apache/doris/regression/ConfigOptions.groovy [hunk 3]
```diff
@@ -141,6 +162,15 @@ class ConfigOptions {
                 .longOpt("directories")
                 .desc("only the use cases in these directories can be executed")
                 .build()
+        excludeDirectoriesOpt = Option.builder("xd")
+                .argName("directories")
+                .required(false)
+                .hasArg(true)
+                .optionalArg(true)
+                .type(String.class)
+                .longOpt("directories")
```
[the remark this answers is not in the document]
Comment (924060929) [reply regression-test/framework/src/main/groovy/org/apache/doris/regression/ConfigOptions.groovy]: done

File: run-regression-test.sh (hunk 1 above)
Comment (pingchunzhang) [reply run-regression-test.sh:43]: ok

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]

**Pull request merged**